### ISSUE TITLE
SQLite library in TDLib conflicts to FCM' SQLite library

### ISSUE THREAD

**Issue by k0ry** [tdlib/td#2003]
Hello,
on iOS projects there is conflict with SQLite library in TDLib and Googles Firebase (FCM)' SQLite library, causing runtime error:
onError Error(code: 400, message: "[query:SELECT sqlcipher_export(\'encrypted\')] to database \"
/Library/Caches/tdlib/db.sqlite\" failed: no such function: sqlcipher_export")

How to resolve this issue?

**Comment by levlam** [comment]
Are FCM and TDLib built as different dylib libraries and linked dynamically?

**Comment by d99kris** [comment]
Stumbled across this issue report and just wanted to mention that I too encountered an sqlite3 conflict between tdlib and another component (in my case sqlite3 from go). I had already modularized my code into shared libs, but in order to resolve the issue I had to switch to library loading at runtime, rather than linking them directly. For my project it resulted in the following commit https://github.com/d99kris/nchat/commit/fd0709c3 but the essence was to use `dlopen()` rather than linking at build time. Just wanted to share in case it's of any help. I have no iOS experience, so it might be completely inapplicable.
